oauthenticator/oauth2.py

```diff
@@ -115,7 +117,22 @@ def get(self):
         state_id = self._generate_state_id()
         next_url = self._get_next_url()
 
-        cookie_state = _serialize_state({"state_id": state_id, "next_url": next_url})
+        state = {"state_id": state_id, "next_url": next_url}
+
+        if self.authenticator.require_pkce:
+            # https://datatracker.ietf.org/doc/html/rfc7636#section-4
+            code_verifier = secrets.token_urlsafe(43)
```

Comment:
FWIW, the spec [recommends](https://datatracker.ietf.org/doc/html/rfc7636#section-4.1) a 32-byte token. The input to `secrets.token_*` is the number of random bytes not the encoded length, so with

```suggestion
            code_verifier = secrets.token_urlsafe(32)
```

`len(code_verifier)` will be 43.